proposals/Exceptions.md

```diff
@@ -280,6 +280,30 @@ calls until a corresponding, enclosing try block is found. It may also associate
 a stack trace that can be used to report uncaught exceptions. However, the
 details of this is left to the embedder.
 
+### Traps and JS API
+
+The `catch` instruction catches exceptions generated by the `throw` instruction,
+but does not catch traps. The rationale for this is that in general traps are
+not locally recoverable and are not needed to be handled in locals scopes like
+try-catch.
+
+The `catch` instruction catches foreign exceptions generated from calls to
+function imports as well, including JavaScript exceptions, with a few
+exceptions:
+1. In order to be consistent before and after a trap reaches a JavaScript frame,
+   the `catch` instruction does not catch exceptions generated from traps. This
+   is currently
+   [`WebAssembly.RuntimeError`](https://webassembly.github.io/reference-types/js-api/#exceptiondef-runtimeerror)
```

Comment:
I suggest to be even more explicit here:
These are currently instances of [`WebAssembly.RuntimeError`](...), but this detail is not used to decide whether to catch the exception or not. Implementations are supposed to specially mark non-catchable exceptions.

Reply:
Reworded using the phrases you suggested.